File: BeatOn/ClientModels/ClientChangeColor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using QuestomAssets.AssetOps;
using QuestomAssets.Models;

namespace BeatOn.ClientModels
{
    [Message(MessageType.ChangeColor)]
    public class ClientChangeColor : MessageBase
    {
        public BeatSaberColor Color { get; set; }

        public ColorType ColorType { get; set; }
    }

}

File: frontend/src/app/models/ClientChangeColor.ts
import { MessageBase, MessageType } from './MessageBase';
import { BeatSaberColor } from './BeatSaberColor';

export class ClientChangeColor extends MessageBase {
    constructor() {
        super();
    }
    readonly Type: MessageType = MessageType.ChangeColor;
    Color: BeatSaberColor;

    ColorType: ColorType;
}
export enum ColorType {
    LeftColor = <any>'LeftColor',
    RightColor = <any>'RightColor',
}
